Core.Auditing/IfxAuditLogger.cs

```diff
@@ -61,36 +60,46 @@ public void LogApplicationAuditEvent(ITelemetryClient telemetryClient, AuditMand
 
         public void LogTokenGenerationAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, TargetResource[] targetResources, CallerIdentity[] callerIdentities, string tokenType, AuditOptionalProperties auditOptionalProperties = null)
         {
-            LogAuthorizationAuditEvent(telemetryClient, operationResult, tokenType + TokenGenerationOperation, targetResources, callerIdentities, auditOptionalProperties);
+            AuditMandatoryProperties auditMandatoryProperties = new AuditMandatoryProperties()
+            {
+                ResultType = operationResult,
+                OperationName = tokenType + TokenGenerationOperation,
+            };
+            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Authorization);
+            LogAuditEvent(telemetryClient,  targetResources, callerIdentities, auditMandatoryProperties, auditOptionalProperties);
         }
 
         public void LogCertificateFetchAuditEvent(ITelemetryClient telemetryClient, OperationResult operationResult, TargetResource[] targetResources, CallerIdentity[] callerIdentities, AuditOptionalProperties auditOptionalProperties = null)
         {
-            LogAuthorizationAuditEvent(telemetryClient, operationResult, FetchCertificateOperation, targetResources, callerIdentities, auditOptionalProperties);
+            AuditMandatoryProperties auditMandatoryProperties = new AuditMandatoryProperties()
+            {
+                ResultType = operationResult,
+                OperationName = FetchCertificateOperation
+            };
+            auditMandatoryProperties.AddAuditCategory(AuditEventCategory.Authorization);
+            LogAuditEvent(telemetryClient, targetResources, callerIdentities, auditMandatoryProperties, auditOptionalProperties);
```

Comment:
This is almost identical (except OperationName, which can be parametrized) with the above. Could you please check again?